lib/puppet/face/config.rb

```diff
@@ -56,6 +57,13 @@
 
     when_invoked do |*args|
       options = args.pop
+
+      @default_section = false
+      if (options[:section].is_a? Array)
```

Comment:
I was thinking of creating a sentinel like:

```ruby
MAIN_SECTION = Object.new
...
default_to { MAIN_SECTION }
...
@default_section = false
if options[:section] == MAIN_SECTION
  options[:section]  = "main"
  @default_section = true
end
```

Reply:
Yes, although I don't like sentinels when user input can be arbitrary.